meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,757 +10,1031 @@ import {
+												studio={studio}
+												translationNamespaces={translationNamespaces}
+												studioMappings={studioMappings}
+												toggleExpanded={toggleExpanded}
+												isExpanded={isExpanded(routeSet.id)}
+												overrideHelper={overrideHelper}
+												exclusivityGroupsFromOverrides={exclusivityGroupsFromOverrides}
+											/>
+										) : (
+											<RenderRouteSetDeletedEntry routeSet={routeSet} overrideHelper={overrideHelper} />
+										)}
+									</React.Fragment>
+								)
+							})}
+						</tbody>
+					</table>
+					<div className="mod mhs">
+						<button className="btn btn-primary" onClick={addNewRouteSet}>
+							<FontAwesomeIcon icon={faPlus} />
+						</button>
+					</div>
+				</React.Fragment>
+			)}
+		</div>
+	)
+}
+
+interface IRenderRouteSetProps {
+	routeSet: WrappedOverridableItemNormal<StudioRouteSet>
+	manifest: MappingsSettingsManifests
+	studio: DBStudio
+	translationNamespaces: string[]
+	studioMappings: ReadonlyDeep<MappingsExt>
+	toggleExpanded: (layerId: string, force?: boolean) => void
+	isExpanded: boolean
+	overrideHelper: OverrideOpHelper
+	exclusivityGroupsFromOverrides: WrappedOverridableItem<StudioRouteSetExclusivityGroup>[]
+}
+
+function RenderRouteSet({
+	routeSet,
+	manifest,
+	studio,
+	translationNamespaces,
+	toggleExpanded,
+	isExpanded,
+	studioMappings,
+	overrideHelper,
+	exclusivityGroupsFromOverrides,
+}: Readonly<IRenderRouteSetProps>): React.JSX.Element {
+	const { t } = useTranslation()
+	const toggleEditRouteSet = React.useCallback(() => toggleExpanded(routeSet.id), [toggleExpanded, routeSet.id])
+
+	const confirmRemove = (routeSetId: string) => {
+		doModalDialog({
+			title: t('Remove this Route Set?'),
+			yes: t('Remove'),
+			no: t('Cancel'),
+			onAccept: () => {
+				overrideHelper.deleteItem(routeSetId)
+			},
+			message: (
+				<React.Fragment>
+					<p>{t('Are you sure you want to remove the Route Set "{{routeId}}"?', { routeId: routeSetId })}</p>
+					<p>{t('Please note: This action is irreversible!')}</p>
+				</React.Fragment>
+			),
+		})
+	}
+
+	const addNewRouteInSet = (routeId: string) => {
+		const newRoutes = routeSet.computed?.routes || []
+
+		newRoutes.push({
+			mappedLayer: '',
+			outputMappedLayer: '',
+			remapping: {},
+			routeType: StudioRouteType.REROUTE,
+		})
+
+		overrideHelper.setItemValue(routeId, 'routes', newRoutes)
+	}
+
+	const updateRouteSetId = React.useCallback(
+		(newRouteSetId: string) => {
+			overrideHelper.changeItemId(routeSet.id, newRouteSetId)
+			toggleExpanded(newRouteSetId, true)
+		},
+		[overrideHelper, toggleExpanded, routeSet.id]
+	)
+
+	const exclusivityGroupOptions = React.useMemo(() => {
+		return getDropdownInputOptions([
+			{
+				name: 'None',
+				value: undefined,
+			},
+			...exclusivityGroupsFromOverrides
+				.filter((group) => group.type === 'normal')
+				.map((group) => group.computed?.name || group.id),
+		])
+	}, [studio.routeSetExclusivityGroupsWithOverrides])
```

Comment:
This dependency looks wrong, as that value isnt used internally by `exclusivityGroupsFromOverrides` is.
The side effect of this being wrong is that whenever `exclusivityGroupsFromOverrides` changes, it wouldn't cause this `exclusivityGroupOptions` value to be recalculated  (unless `studio.routeSetExclusivityGroupsWithOverrides` also changed)
```suggestion
	}, [exclusivityGroupsFromOverrides])
```